beets/library.py

```diff
@@ -1601,14 +1603,21 @@ class Library(dbcore.Database):
     def __init__(
         self,
         path="library.blb",
-        directory="~/Music",
+        directory: Optional[str] = None,
         path_formats=((PF_KEY_DEFAULT, "$artist/$album/$track $title"),),
         replacements=None,
     ):
         timeout = beets.config["timeout"].as_number()
         super().__init__(path, timeout=timeout)
 
-        self.directory = bytestring_path(normpath(directory))
+        if directory is not None:
+            self.directory = normpath(directory)
+        else:
+            # Check if the user specified something in the XDG user dirs.
+            # Otherwise, fall back to '~/Music'.
+            music_dir = platformdirs.user_music_path()
+            self.directory = bytes(music_dir)
```

Comment:
```python
            # Check if the user specified something in the XDG user dirs.
            # Otherwise, fall back to '~/Music'.
            music_dir = platformdirs.user_music_path()
            self.directory = bytes(music_dir)
```
I found the comment above slightly confusing since the logic below it did not seem to perform any checks. Consider below

```suggestion
        # If the user has not configured a directory, fall back to '~/Music'
        if directory is None:
            directory = platformdirs.user_music_path()

        self.directory = bytestring_path(directory)
```

Reply:
Right, I carried that over from the previous PR.  Will fix.